### ISSUE TITLE
Improve translation quality of English to Chinese

### ISSUE THREAD

**Issue by eu9ene** [mozilla/translations#1033]
The quality of the teacher model doesn't look great: COMET 85.67 vs Google Translate 89.45 (-4.41%) on flores-test. 

I suspect we have some issues in the training data:
1. I found that we used dataset `mtdata_Statmt-backtrans_enzh-wmt20-eng-zho` which is backtranslations for zh-en WMT20 task. So the zh part of it might be of poor quality. We already have our own back-translations.
2. We convert all the data from Traditional to Simplified with hanzi-convert. People on the internet say such conversion is imprecise due to the regional differences, vocabulary, word composition etc., so it is not equal to a proper translation. It might be the case that it pollutes data when Chinese is the target language. We can just filter all the data in Traditional and see if it makes a difference.
3. We use a 64k joint vocabulary for very different scripts, it's worth running an experiment with separate vocabularies 32k in size. Depends on https://github.com/mozilla/translations/issues/913
4. There are known issues for some datasets: [UN](https://github.com/mozilla/translations/issues/744), [HPLT](https://github.com/mozilla/translations/issues/1030), [desegmentation](https://github.com/mozilla/translations/issues/743). We can address them with ad-hoc fixes through OpusCleaner and the mono-cleaning scripts.

The question is which of those issues contribute most. It would require running separate experiments for most of them to see if it moves the needle.

Another indicator that something is wrong with the data is that one of my experiments completely diverged and I had to reduce the learning rate. Also when training in two stages there is a big bump in the cost when moving to training on the original parallel corpus only. This typically indicates that the fine-tuning data is noisier compared to the pre-training one (a mix of back-translated and original corpus).

<img width="625" alt="Image" src="https://github.com/user-attachments/assets/87515284-bbbc-43f7-b42d-eaf9d01a58eb" />

**Comment by eu9ene** [comment]
Looking at a UN [sample](https://opus.nlpl.eu/sample/en&zh/UNPC&v1.0/sample), the numbers don't match at all... We should either return the num_mismatch filter for it or exclude the UN corpus completely. cc @ZJaume

**Comment by ZJaume** [comment]
If you mean the number prefixes (`14:1`), they are just identifiers from the opus xml internal format, not part of the corpus: https://opus.nlpl.eu/legacy/UNPC/v1.0/en-zh_sample.html

**Comment by ZJaume** [comment]
I suggested [OpenCC](https://pypi.org/project/OpenCC/) as it is much more recent (hanziconv last release was in 2016) and seems widely used. But, I do not know if it would solve the issues that people is claiming about.

**Comment by eu9ene** [comment]
> If you mean the number prefixes (`14:1`), they are just identifiers from the opus xml internal format, not part of the corpus: https://opus.nlpl.eu/legacy/UNPC/v1.0/en-zh_sample.html

Interesting, I don't see them in [NLLB](https://opus.nlpl.eu/sample/en&zh/NLLB&v1/sample) for example.

**Comment by eu9ene** [comment]
> I suggested [OpenCC](https://pypi.org/project/OpenCC/) as it is much more recent (hanziconv last release was in 2016) and seems widely used. But, I do not know if it would solve the issues that people is claiming about.

Actually, I use opencc to convert and hanzi-conv only to identify the script.

**Comment by eu9ene** [comment]
Training en-zh with separate vocabs on the same dataset looks consistently better. The difference is not as significant as after fixing the dataset issues

<img width="1724" alt="Image" src="https://github.com/user-attachments/assets/78b55ce2-d495-4150-a611-9f6cbad4fb1d" />
